Source/SA/DPD/DPD.cs

```diff
@@ -0,0 +1,207 @@
+                    MemoryPolynomialLeadOrder = 2,
+                    MemoryPolynomialLagOrder = 2,
+                    MemoryPolynomialLeadMemoryDepth = 2,
+                    MemoryPolynomialLagMemoryDepth = 2,
+                    MemoryPolynomialMaxLead = 2,
+                    MemoryPolynomialMaxLag = 2
+                };
+            }
+        }
+
+        public struct LookupTableResults
+        {
+            public ComplexWaveform<ComplexSingle> WaveformWithDpdComplexSingle;
+            public float[] LookupTableInputPowers;
+            public ComplexSingle[] LookupTableComplexGains;
+            public double PostDPDPAPR;
+            public double PowerOffset;
+        }
+
+        public struct MemoryPolynomialResults
+        {
+            public ComplexWaveform<ComplexSingle> WaveformWithDpdComplexSingle;
+            public ComplexSingle[] DpdPolynomial;
+            public double PostDPDPAPR;
+            public double PowerOffset;
+        }
+        #endregion
+        #region ConfigureDPD
+        public static void ConfigureCommon(RFmxSpecAnMX specAnSignal, CommonConfiguration commonConfig, Waveform waveform, string selectorString = "")
+        {
+            RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent idlePresent;
+     
+            idlePresent = waveform.IdleDurationPresent ? RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent.True : RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent.False;
+
+            specAnSignal.SelectMeasurements(selectorString, RFmxSpecAnMXMeasurementTypes.Dpd, true);
+
+            specAnSignal.Dpd.Configuration.ConfigureReferenceWaveform(selectorString, waveform.WaveformData, idlePresent, commonConfig.SignalType);
+            specAnSignal.Dpd.Configuration.ConfigureDutAverageInputPower(selectorString, commonConfig.DutAverageInputPower);
+            specAnSignal.Dpd.Configuration.ConfigureMeasurementInterval(selectorString, commonConfig.DpdMeasurementInterval);
+            specAnSignal.Dpd.Configuration.ConfigureMeasurementSampleRate(selectorString, commonConfig.DpdSampleRateMode, commonConfig.SampleRate);
+
+        }
+
+        public static void ConfigureRFmxLUT(RFmxSpecAnMX specAnSignal, LookupTableConfiguration lutConfig, string selectorString = "")
+        {
+            specAnSignal.Dpd.Configuration.ConfigureDpdModel(selectorString, RFmxSpecAnMXDpdModel.LookupTable);
+            specAnSignal.Dpd.Configuration.ConfigureLookupTableType(selectorString, lutConfig.LookupTableType);
+            specAnSignal.Dpd.Configuration.ConfigureLookupTableThreshold(selectorString, lutConfig.ThresholdEnabled, lutConfig.ThresholdLevel,
+                lutConfig.ThresholdType);
+            specAnSignal.Dpd.Configuration.ConfigureLookupTableStepSize(selectorString, lutConfig.LookupTableStepSize);
+        }
+
+        public static void ConfigureRFmxMP(RFmxSpecAnMX specAnSignal, MemoryPolynomialConfiguration mpConfig, string selectorString = "")
```

Comment:
As above, I think that this should be 
```suggestion
        public static void ConfigureMemoryPolynomial(RFmxSpecAnMX specAnSignal, MemoryPolynomialConfiguration mpConfig, string selectorString = "")
```

Reply:
@mdwhitten agreed

Reply:
@cindyxie why did you add "DPD" to the end? Suggest removing that.